packages/drivers/debugger/src/fluidDebuggerController.ts

```diff
@@ -310,3 +337,39 @@ export class DebugReplayController extends ReplayController implements IDebugger
         this.startSeqDeferred.resolve(seq);
     }
 }
+
+async function* generateSequencedMessagesFromDeltaStorage(deltaStorage: IDocumentDeltaStorageService)  {
+    const timeStart = Date.now();
+    let requests = 0;
+    let opsStorage = 0;
+    let lastSeq = 0;
+    const batch = 2000;
+    while (true) {
+        requests++;
+        const messages = await loadChunk(lastSeq, lastSeq + batch, deltaStorage);
+        if (messages.length === 0) {
+            break;
+        }
+        yield messages;
+        opsStorage += messages.length;
+        lastSeq = messages[messages.length - 1].sequenceNumber;
+    }
+
+    if (requests > 0) {
+        // eslint-disable-next-line max-len
+        console.log(`\n${Math.floor((Date.now() - timeStart) / 1000)} seconds to retrieve ${opsStorage} ops in ${requests} requests`);
+    }
+}
+
+async function loadChunk(from: number, to: number, deltaStorage: IDocumentDeltaStorageService) {
+    console.log(`Loading ops at ${from}`);
+    for (let iter = 0; iter < 3; iter++) {
+        try {
+            return await deltaStorage.get(from, to);
+        } catch (error) {
+            console.error("Hit error while downloading ops. Retrying");
+            console.error(error);
```

Comment:
That's rather critical error, and messages will have a gap. I think whole download process should fail (i.e. this code should throw)